scripts/tests/py/metadata.py

```diff
@@ -0,0 +1,160 @@
+# Copyright (c) 2023 Project CHIP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from dataclasses import dataclass
+from pathlib import Path
+import re
+import yaml
+
+
+@dataclass
+class Metadata:
+    py_script_path: str = None
+    run: str = None
+    app: str = None
+    factoryreset: bool = False
+    comissioning_method: str = None
+    discriminator: int = None
+    kvs: str = None
+    storage_path: str = None
+    on_network_commission: str = None
+    passcode: int = None
+    trace_to: str = None
+    hex_arg: str = None
+    endpoint: str = None
+    manual_code: str = None
+    bool_arg: str = None
+    PICS: str = None
+    tests: str = None
+    int_arg: str = None
+    trace_to_appjson: str = None
+    trace_to_testjson: str = None
+    trace_toperfetto: str = None
+
+    def print(self):
+        print(repr(self))
+
+
+class Metadata_Reader:
+    # This class will be initialized with
+    # the name of the script folder and the environment
+    # file name
+    
+    def __init__(self, script_folder, env_file_name):
+        self.script_folder = script_folder
+        self.env = self.__build_env_object__(env_file_name)
+
+
+    # builds the environment object 
```

Comment:
A method named "build_env_object" obviously `Builds the environment object`.
Please update comments for:
  - use python doc comments (i.e. strings after the declaration of the method)
  - add info in them that is not just re-stating the signature. For example reading this signature I could write `This function builds the environment object. It operates on self object and takes in the environment file name as an argument` which feels like a lot however it is 100% filler text without any help to the code reader.

for documentation you could have things like:
  - file name is a path to a YAML file (maybe the name should be env_yaml_file_path)
  - describe what an environment object is or where to find documentation about it